src/util/workers.c

```diff
@@ -30,33 +31,58 @@ static void yieldCallback(void *yieldCtx) {
   RedisModule_Yield(ctx, REDISMODULE_YIELD_FLAG_CLIENTS, NULL);
 }
 
+/* Configure here anything that needs to know it can use the thread pool */
+static void workersThreadPool_OnTurnOn(size_t new_num) {
+  // Log that we've enabled the thread pool.
+  RedisModule_Log(RSDummyContext, "notice", "Enabled workers threadpool of size %lu", new_num);
+  // Change VecSim write mode temporarily for fast RDB loading of vector index (if needed).
+  VecSim_SetWriteMode(VecSim_WriteAsync);
+}
+
+/* Configure here anything that needs to know it cannot use the thread pool anymore */
+static void workersThreadPool_OnTurnOff(size_t old_num) {
+  RedisModule_Log(RSDummyContext, "notice", "Disabled workers threadpool of size %lu", old_num);
+  VecSim_SetWriteMode(VecSim_WriteInPlace);
+}
+
 // set up workers' thread pool
 int workersThreadPool_CreatePool(size_t worker_count) {
   assert(_workers_thpool == NULL);
 
   _workers_thpool = redisearch_thpool_create(worker_count, RSGlobalConfig.highPriorityBiasNum, LogCallback, "workers");
   if (_workers_thpool == NULL) return REDISMODULE_ERR;
-
+  if (worker_count > 0) {
+    workersThreadPool_OnTurnOn(worker_count);
+  } else {
+    workersThreadPool_OnTurnOff(worker_count);
+  }
   return REDISMODULE_OK;
 }
 
-size_t workersThreadPool_SetNumWorkers(size_t worker_count) {
-  if(_workers_thpool == NULL) return worker_count;
+void workersThreadPool_SetNumWorkers() {
+  if (_workers_thpool == NULL) return;
 
+  size_t worker_count = RSGlobalConfig.numWorkerThreads;
+  if (in_event && RSGlobalConfig.minOperationWorkers > worker_count) {
```

Comment:
Don't we need here ()? (no side effects to determine the order)
```suggestion
  if (in_event && (RSGlobalConfig.minOperationWorkers > worker_count)) {
```

Reply:
Arithmetic operations and relations precede logic operations. [Nice table here](https://learn.microsoft.com/en-us/cpp/c-language/precedence-and-order-of-evaluation?view=msvc-170#precedence-and-associativity-of-c-operators)